scripts/gc.js

```diff
@@ -0,0 +1,433 @@
+      }
+    }  
+    console.log(`${i}, ${leaf_num}, ${leaf_size}, ${branch_num}, ${branch_size}, ${ext_num}, ${ext_size}, ${arr_num}, ${arr_size}, ${hdf.change.size}`);
+    hdi = new memwatch.HeapDiff();
+  }
+
+  console.log("Tree removal");
+  console.log("Batch, ", "Leafs, ", "Leaf nodes size, ", "Branches, ", "Branch nodes size, ", "Extensions, ", "Extension nodes size, ", "Array elements diff, ", "Array elements size, ", "Total Size");
+
+  hdi = new memwatch.HeapDiff();
+  for (let i = 0; i < maxBatches; i++) {
+    trees.shift();
+    global.gc();
+    hdf = hdi.end();
+    let arr_num, arr_size, branch_num, branch_size, leaf_num, leaf_size, ext_num, ext_size;
+    let nodels = hdf.change.details;
+    for (let node of nodels) {
+      if (node["what"] == 'Array') {
+        arr_num = node["+"] - node["-"];
+        arr_size = node.size;
+      }
+      if (node["what"] == 'LeafNode') {
+        leaf_num = node["+"] - node["-"];
+        leaf_size = node.size;
+      }
+      if (node["what"] == 'BranchNode') {
+        branch_num = node["+"] - node["-"];
+        branch_size = node.size;
+      }
+      if (node["what"] == 'ExtensionNode') {
+        ext_num = node["+"] - node["-"];
+        ext_size = node.size;
+      }
+    }
+    console.log(`${i}, ${leaf_num}, ${leaf_size}, ${branch_num}, ${branch_size}, ${ext_num}, ${ext_size}, ${arr_num}, ${arr_size}, ${hdf.change.size}`);
+    hdi = new memwatch.HeapDiff();
+  
+  }
+}
+
+let state_global;
+let stats_global = [];
+let flag = 0;
+
+const generateStats = function() {
+  if (flag == 7) {
+    state_global = null;
+  }
+
+  flag = flag + 1;
+  let heapUsed = process.memoryUsage().heapUsed;
+  stats_global.push(heapUsed);
+}
+
+const test_memgraph_ETH_batch = (els) => {
+  
+  console.log("Tree creation for ETH batch");
+  const maxKeys = process.argv[3];
+  state_global = new ethereumjs();
+
+  let flag_ETH = false;
+  state_global.batch(els, () => {flag_ETH = true});
+  wait.for.predicate(() => flag_ETH);
+
+  setInterval(generateStats, 1000);
+
+  process.on('SIGINT', function(){
+    var data = JSON.stringify(stats_global);
+    fs.writeFile("./graph/stats_ETH_batch_" + maxKeys + ".json", data, function(err) {
+      if(err) {
+        console.log(err);
+      } else {
+        flag = 0;
+        console.log("\nSaved stats");
+      }
+      process.exit();
+    });
+  });
+}
+const test_memgraph_RBC_batch = (els) => {
+  
+  console.log("Tree creation for batch");
+  const maxKeys = process.argv[3];
+  state_global = new rainblock.MerklePatriciaTree();
+  state_global.batch(els);
+
+  setInterval(generateStats, 1000);
+
+  process.on('SIGINT', function(){
+    var data = JSON.stringify(stats_global);
+    fs.writeFile("./graph/stats_RBC_batch_" + maxKeys + ".json", data, function(err) {
+      if(err) {
+        console.log(err);
+      } else {
+        flag = 0;
+        console.log("\nSaved stats");
+      }
+      process.exit();
+    });
+  });
+}
+
```

Comment:
Could we have testMemGraphBatch( els, state ) and not have two functions separately for RBC and ETH?